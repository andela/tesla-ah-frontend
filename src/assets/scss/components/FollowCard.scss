.follow-card {
  background-color: #fff;
  padding: 10px;
  margin-bottom: 10px;
}

.follow-card .follow-card-content {
  margin-left: 10px;
}

.follow-card .follow-card-info {
  margin-bottom: 10px;
}

.follow-card .btn-custom {
  width: 150px;
  border-radius: 30px !important;
}

.follow-card .btn-custom.following {
  color: #C4C4C4;
  background: #fff;
  border: 1px solid #C4C4C4;
}

.follow-card .d-avatar {
  height: 80px;
  width: 80px;
}
