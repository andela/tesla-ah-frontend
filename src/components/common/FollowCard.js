import React from 'react';
import { capitalize } from 'lodash';

import '../../assets/scss/components/FollowCard.scss';

const FollowCard = (props) => {
  const {
    avatar,
    firstName,
    lastName,
    username,
    noButton,
    bio,
  } = props;

  return (
    <React.Fragment>
      <div className="follow-card d-flex">
        <img className="d-avatar" src={avatar} alt="avatar" />
        <div className="follow-card-content flex-grow-1">
          <div className="d-flex justify-content-between align-items-center">
            <div className="follow-card-info d-flex flex-column align-items-start">
              <span className="fullname">
                {`${capitalize(firstName)} ${capitalize(lastName)}`}
              </span>
              <span className="username">{`@${username}`}</span>
            </div>
            {/* <button type="button" className="btn-custom following"> */}
            {!noButton ? (
              <button type="button" className="btn-custom">
                Follow
              </button>
            ) : null}
          </div>
          <div className="bio">{bio}</div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default FollowCard;
