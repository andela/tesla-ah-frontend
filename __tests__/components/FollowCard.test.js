/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import FollowCard from '../../src/components/common/FollowCard';
import { users } from '../mockData';

const renderFollowCard = (args) => {
  const defaultProps = {
    avatar: '',
    firstName: '',
    lastName: '',
    username: '',
    noButton: null,
  };
  const props = { ...defaultProps, ...args };

  return shallow(<FollowCard {...props} />);
};

it('renders FollowCard component', () => {
  const wrapper = renderFollowCard();
  expect(wrapper).toMatchSnapshot();
});

it('renders FollowCard component with follow button', () => {
  const {
    avatar,
    firstName,
    lastName,
    username,
  } = users[0];
  const wrapper = renderFollowCard({
    avatar,
    firstName,
    lastName,
    username,
  });
  expect(wrapper.find('.follow-card').length).toBe(1);
  expect(wrapper.find('.btn-custom').length).toBe(1);
});

it('renders FollowCard component without follow button', () => {
  const {
    avatar,
    firstName,
    lastName,
    username,
  } = users[0];
  const wrapper = renderFollowCard({
    avatar,
    firstName,
    lastName,
    username,
    noButton: true,
  });
  expect(wrapper.find('.follow-card').length).toBe(1);
  expect(wrapper.find('.btn-custom').length).toBe(0);
});
