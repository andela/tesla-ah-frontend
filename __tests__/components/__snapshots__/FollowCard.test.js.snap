// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FollowCard component 1`] = `ShallowWrapper {}`;
